### PR TITLE
 Updated to only show campusses when provider has none and also replaced slider with add + remove buttons (ford3-162)

### LINKED ISSUE #162: Replace the campus add slider with add/remove campus functionality

### PR DESCRIPTION
**Pull request by Christiaanvdm**
This closes #162 
NewGIF

### CODE REVIEW

File: django_project/ford3/static/js/provider_form.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-slider.oninput = function () {
-    output.value = this.value;
-
-    createCampusNameInputs(this.value);
+output.onchange = function () {
+    createCampusNameInputs(output.value);
 }
 
+$(document).on("submit", "form", function(e) {
+    var oForm = $(this);
```
Comment (pierrealixt) [inline django_project/ford3/static/js/provider_form.js]: when are you using `oForm`?

File: django_project/ford3/static/js/provider_form.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 var output = document.getElementById('number-of-campuses');
-slider.value = 1;
-output.value = slider.value; // Display the default slider value
 // Update the current slider value (each time you drag the slider handle)
-slider.oninput = function () {
-    output.value = this.value;
-
-    createCampusNameInputs(this.value);
+output.onchange = function () {
+    createCampusNameInputs(output.value);
```
Comment (pierrealixt) [inline django_project/ford3/static/js/provider_form.js]: What about your idea of having only one campus input text and a button `add campus` that will add a new input text.

File: django_project/ford3/static/js/provider_form.js [hunk 3]
```diff
@@ -1,34 +1,44 @@
...
+       return true;
+    }
+    else {
+       return false;
+    }
+})
 
+$('#add-campus-name').click(
+    function() {addCampusNameInput()
+    $('.remove-campus-button').click(function() {
```
Comment (pierrealixt) [inline django_project/ford3/static/js/provider_form.js]: Do not repeat yourself
See line 1 to 4 from this same file

File: django_project/ford3/views/provider.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                return render(request, 'provider_form.html', {'form': form})
+            redirect_url = '/providers/' + str(new_provider.id)
+            return redirect(redirect_url)
+    else:
+        provider = get_object_or_404(
+            Provider,
+            id=provider_id
+        )
+        form = ProviderForm(instance=provider)
+        if len(provider.campus) > 0:
```
Comment (pierrealixt) [inline django_project/ford3/views/provider.py]: `is_new_provider = len(provider.campus) > 0`

Comment (pierrealixt) [reply django_project/ford3/views/provider.py]: This could be a method of the Provider model

Comment (Christiaanvdm) [reply django_project/ford3/views/provider.py]: The thought crossed my mind but I am worried that is pre-optimising?

Comment (Christiaanvdm) [reply django_project/ford3/views/provider.py]: Agreed. Thanks!

File: django_project/ford3/static/js/provider_form.js [hunk 5]
```diff
@@ -14,14 +20,14 @@ $(document).on("submit", "form", function(e) {
...
 
 $('#add-campus-name').click(
-    function() {addCampusNameInput()
-    $('.remove-campus-button').click(function() {
-        this_button_parrent_parent = $(this).parent().parent();
-        this_button_parrent_parent.remove();
-    })
+    function () {
+        addCampusNameInput();
+        innitiate_remove_campus_buttons();
```
Comment (pierrealixt) [inline django_project/ford3/static/js/provider_form.js]: in JS, we prefer camelCase style: `initiateRemoveCampusButtons`

**Comment by pierrealixt** [comment]
LGTM

**Review by pierrealixt — APPROVED** [review]
LGTM

**Pull request merged**